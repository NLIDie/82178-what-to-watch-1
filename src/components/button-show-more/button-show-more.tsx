import * as React from 'react';

// Types
import { TProps } from './button-show-more.types';

export function ButtonShowMore(props: TProps): JSX.Element {
  const {onClick} = props;

  return (
    <div className="catalog__more">
      <button className="catalog__button" type="button" onClick={onClick}>
        Show more
      </button>
    </div>
  );
}
