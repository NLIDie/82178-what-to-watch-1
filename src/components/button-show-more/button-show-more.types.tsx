import * as React from 'react';

export type TProps = {
  onClick: (event: React.SyntheticEvent<HTMLButtonElement>) => void;
}
