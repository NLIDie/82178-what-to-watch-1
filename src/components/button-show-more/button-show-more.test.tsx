import * as React from "react";
import {create} from "react-test-renderer";

// Components
import {ButtonShowMore} from "./button-show-more";

describe(`Components/ButtonShowMore`, () => {
  it(`Correct renderer the component`, () => {
    const tree = create(<ButtonShowMore onClick={jest.fn()} />);

    expect(tree).toMatchSnapshot();
  });
});
