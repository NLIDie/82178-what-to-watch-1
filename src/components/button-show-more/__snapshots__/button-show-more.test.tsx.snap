// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/ButtonShowMore Correct renderer the component 1`] = `
<div
  className="catalog__more"
>
  <button
    className="catalog__button"
    onClick={[MockFunction]}
    type="button"
  >
    Show more
  </button>
</div>
`;
